### ISSUE TITLE
Hazard method: Expand Point events to include (circular) area in surroundings

### ISSUE THREAD

**Issue by Evelyn-M** [CLIMADA-project/climada_python#126]
**Description of the issue / idea:** At the moment, if one has hazard events that are of point geometry (say, historic landslides that are reported without geographic extent), there is no neat method in the hazard module which allows "expanding" the hazard extent to a certain area around it.

**Status Quo:** To date, there is a workaround in the impact-calculation chain, where one can match the exposure with the hazard centroids, and specify the distance to the centroid until which a surrounding exposure should be attributed to a particular centroid (`exp.assign_centroids()`).

I was wondering wether to expand a similar method directly on the hazard / centroid instance.
Sth of the sort

  ```
 def _incl_affected_surroundings(self, max_dist):
        """
        Change centroids' geometry from POINT to circular POLYGON within 
        given radius.
        
        Parameters
        ----------
            max_dist (int): distance in metres until which
                surroundings of a shapely.Point count as affected
        Returns
        -------
            
        """
        self.centroids.set_geometry_points()
        self.centroids.geometry = self.centroids.geometry.buffer(
            deg_from_dist(max_dist))
    
```

**Comment by chahank** [comment]
This is a very important point to expand the use of CLIMADA. 

Would you agree with the following thought:
What is proposed is to make a hazard agnostic function which allows to expand reported point data to area data? So, make what `TropCyclone.set_from`tracks()` does by transforming tracks into windfields? 

If you think this is the same idea, maybe it is worth trying to see how one could make a general abstract concept which would fit to both use cases landslides and tropical cyclones.

**Comment by Evelyn-M** [comment]
> Would you agree with the following thought:
> What is proposed is to make a hazard agnostic function which allows to expand reported point data to area data? 
Yes, i would. 

> So, make what TropCyclone.set_fromtracks()` does by transforming tracks into windfields, but for 'any' hazard?

I have not looked into how tracks are transformed into wind fields and guess the "expansion" there is much more sophisticated than just taking a fixed distance or shape around  a point. So yes and no in that case.

**Comment by chahank** [comment]
> I have not looked into how tracks are transformed into wind fields and guess the "expansion" there is much more sophisticated than just taking a fixed distance or shape around a point. So yes and no in that case.

Yes, the track module is more sophisticated. The idea is to try to make the basic task as abstract as possible and then see how to proceed. That is why I propose to make a comparison between landslide and tropcyclone to see what could be needed and how much can be generalized. 

On difference is, as you mentioned, that for tropical cyclones, the hazard intensity is not only derived by defining a region around each track point, but by computing specific values of intensity in that region from the different variables (windspeed, pressure, etc...) at the center point. The similarity to the landslide module would be that probably you do not want to simply have a circular region around your reported point to be affected, but for instance to take into account the terrain geography.

**Comment by togeig** [comment]
@chahank I find this a very relevant thought and support the idea of introducing a very general hazard extension method. This might become handy eg for compound events.

**Comment by togeig** [comment]
I find this a very relevant thought and support the idea of introducing a very general extension method that helps to create hazard-specific extensions that can talk to each other. This might be very helpful also for compund events.

Am 28. Januar 2021 17:23:38 MEZ schrieb Chahan Kropf <notifications@github.com>:

**Comment by bguillod** [comment]
While I understand what you want to achieve and clearly see that this woud be very handy, I think we need to be careful here: the way a hazard can be "expanded" spatially is highly dependent on the hazard type and on the exposure type. I think a generic method as part of the hazard class is a bit dangerous. I would favor the idea of having that within hazard sub-classes as I think it would be much safer.

In most cases, I would say that one usually first define its exposure, then build the hazard using the exposure's coordinates as centroids. In that case, this functionality is not needed. However, sometimes we of course rather generate hazard sets on a pre-defined set of centroids (usually, a regular grid) and then use that for any exposure location on the grid, taking the closest centroids' hazard intensity/fraction.

Originally, CLIMADA's main focus was Tropical Cyclones. For TC wind, the hazard field tends to be quite homogeneous spatially (or say, at least, continuous), which means that as long as your centroids grid is of high enough resolution we are happy with the approximation of using the wind speed as calculated for a location a few km away from the exposure's location. This, however, might be quite different for other hazard types (e.g., I assume landslides are very dependent on the local small-scale topography, hence the landslide hazard can be quite different a few km away, for instance depending on whether you are on a mountain ridge or a valley bottom).

Of course, how exposures are defined also might change the way you assign hazard to exposure: Exposures representing grid cells (area sums or averages, e.g. total asset value within a 5x5km cell) might have hazard attributed differently than an actual point location (e.g., a hotel at an exact set of coordinates), except for very smooth hazard such as wind (again!).

In the end, the user must understand and know what they are doing and whether it makes sense. There is no way around it. But we should be careful not to make it too easy to make physical mistakes. So if you go with a generalized assignment methodology you probably need to think that through for all combinations of exposure types and hazard types and see what can be generalized and what cannot be.

**Comment by chahank** [comment]
Thanks for sharing your thoughts, and I agree on the logic, but not necessarily on the conclusions. The idea would be to have for instance a subclass of Hazard, or better a sub-module where all the work of mapping Hazards onto their centroids. This could make the architecture clearer and allow to gather often used methods in one place. For instance, a very basic method to simply add a radius of action (e.g. for a very simple landslide or earthquake hazard).

**Comment by Evelyn-M** [comment]
All good comments.

I guess to proceed, it would be useful to have a list of hazard-types (LS, EQ, ...) and see what would be meaningful expansions in case they were reported as Point-events.

I agree with Benoît that, in case we conclude that simple shapes (such as circles, rectangles, ...) would not be meaningful across several hazard types, but rather that topology etc. needs to be included and influence different types of hazards in different ways, then those expansion methods should go into the respective hazard modules (they can still be consistently labelled, such as `expand_point_hazard_...` )
